### PR TITLE
Add TorchXLA option to benchmark runner

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99505

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -839,6 +842,39 @@ def baselines(models, model_iter_fn, example_inputs, args):
     return result
 
 
+def xla(args, model_iter_fn, model, example_inputs):
+    xla_dev = xm.xla_device(devkind=current_device)
+    model_xla = copy.deepcopy(model).to("cpu").to(device=xla_dev)
```
Comment (voznesenskym) [inline benchmarks/dynamo/common.py:847]: why to cpu first?

File: benchmarks/dynamo/common.py [hunk 2]
```diff
@@ -2189,6 +2232,12 @@ def run(runner, args, original_dir=None):
         )
         experiment = speedup_experiment
         output_filename = "inductor.csv"
+    elif args.xla:
+        (dev,) = args.devices
+        os.environ["PJRT_DEVICE"] = {"cuda": "GPU", "cpu": "CPU"}[dev]
+        torch._dynamo.mark_dynamic = MagicMock()
```
Comment (voznesenskym) [inline benchmarks/dynamo/common.py:2238]: Why?

**Review by voznesenskym — APPROVED** [review]

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (jansel) [reply benchmarks/dynamo/common.py:847]: Throws an error on direct cuda -> xlacuda.

File: benchmarks/dynamo/common.py (hunk 2 above)
Comment (jansel) [reply benchmarks/dynamo/common.py:2238]: Issues with older version of PyTorch.

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (miladm) [reply benchmarks/dynamo/common.py:847]: what's the perf implication of this line?

Comment (jansel) [reply benchmarks/dynamo/common.py:847]: Should be zero since it done once on startup before any performance measurement.

**Pull request closed without merging**